skia/RSkComponentRNSVGSvgView.cpp

```diff
@@ -0,0 +1,180 @@
+  setLengthAttribute(SkSVGAttribute::kX,std::to_string(component.layoutMetrics.frame.origin.x));
+  setLengthAttribute(SkSVGAttribute::kY,std::to_string(component.layoutMetrics.frame.origin.y));
+  setLengthAttribute(SkSVGAttribute::kWidth,std::to_string(component.layoutMetrics.frame.size.width));
+  setLengthAttribute(SkSVGAttribute::kHeight,std::to_string(component.layoutMetrics.frame.size.height));
+
+  svgContainerSize_.set(component.layoutMetrics.frame.size.width,
+                        component.layoutMetrics.frame.size.height);
+
+  //TODO: ("TO BE CONFIRMED: When the SVG dimensions change, it is unclear whether a remove and insert mount instruction will be received \
+  // or an update mount instruction will be received. If an update mount instruction is received,update props will not be called for layout metric changes.");
+
+  return RnsShell::LayerInvalidateAll;
+}
+
+bool RSkComponentRNSVGSvgView::onPrepareToRender(SkSVGRenderContext* ctx) const {
+  auto viewPortRect  = ctx->lengthContext().resolveRect(x_,y_,width_,height_); 
+  auto contentMatrix = SkMatrix::Translate(viewPortRect.x(), viewPortRect.y());
+  auto viewPort      = SkSize::Make(viewPortRect.width(), viewPortRect.height());
+
+  if (!viewBox_.isEmpty()) {
+    // A viewBox overrides the intrinsic viewport.
+    viewPort = SkSize::Make(viewBox_.width(), viewBox_.height());
+    contentMatrix.preConcat(
+      SkMatrix::MakeRectToRect(viewBox_, viewPortRect, SkMatrix::kFill_ScaleToFit));
+  }
+
+  if (!contentMatrix.isIdentity()) {
+    ctx->saveOnce();
+    ctx->canvas()->concat(contentMatrix);
+  }
+
+  if (viewPort != ctx->lengthContext().viewPort()) {
+    ctx->writableLengthContext()->setViewPort(viewPort);
+  }
+  return this->INHERITED::onPrepareToRender(ctx);
+}
+
+void RSkComponentRNSVGSvgView::onSetAttribute(SkSVGAttribute attr, const SkSVGValue& attrValue) {
+  switch (attr) {
+    case SkSVGAttribute::kX:
+      if (const auto* x = attrValue.as<SkSVGLengthValue>()) {
+        x_ = *x;
+      }
+    break;
+    case SkSVGAttribute::kY:
+      if (const auto* y = attrValue.as<SkSVGLengthValue>()) {
+        y_ = *y;
+      }
+    break;
+    case SkSVGAttribute::kWidth:
+      if (const auto* w = attrValue.as<SkSVGLengthValue>()) {
+        width_ = *w;
+      }
+    break;
+    case SkSVGAttribute::kHeight:
+      if (const auto* h = attrValue.as<SkSVGLengthValue>()) {
+        height_ = *h;
+      }
+    break;
+    case SkSVGAttribute::kViewBox:
+      if (const auto* vb = attrValue.as<SkSVGViewBoxValue>()) {
+        viewBox_=*vb;
+      }
+    break;
+    default:
+      this->INHERITED::onSetAttribute(attr, attrValue);
+  }
+}
+
+inline void RSkComponentRNSVGSvgView::alterSkiaDefaultPaint() {
+//Note: Skia's default paint is black. WhereIn on SVG REference platform it's transparent
+//      To Match with Reference output, for the next immendiate child of SVGView,
+//      resetting color to transparent, if color not speciifed
+
```

Comment:
Cant we change this color during parsing/adding of the component itself

Reply:
Can't due in parsing foe below two reasons 

1. Color & other common renderable Props inherits props from parent's , if it don't have on it's own. We use SkSVG engine for Common renderable props parsing & managing inheritance . If we apply during parsing, then skia's svg engine trat it as property defined and won't apply inheritence.
2. It is Applied only for the Outer most group So , Doing it once all child svg's child insert completes.